### PR TITLE
Feature/dtonbwtwo 426 consumes tax status to automate payment decisions

### PR DESCRIPTION
**Pull request by saikgupta**
This PR is for the java part of the SDK changes. The corresponding ticket is:
https://engineering.paypalcorp.com/jira/browse/DTONBWTWO-426
Following changes were made:
1. Added new variable to HyperwalletUser class.
2. Added new variable to HyperwalletUserListPagination  class.
3. changes done to HyperwalletUserTest class/HyperwalletUserListPaginationTest class.
4. Added Unit Test and IT Test.

### CODE REVIEW

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java [hunk 1]
```diff
@@ -14,27 +14,26 @@
 @XmlAccessorType(XmlAccessType.FIELD)
 public class HyperwalletUser extends HyperwalletBaseMonitor {
 
-    public static enum Gender {MALE, FEMALE}
 
-    public static enum ProfileType {INDIVIDUAL, BUSINESS, UNKNOWN}
+    public enum Gender {MALE, FEMALE}
```
Comment (hrandhawa-hw) [inline src/main/java/com/hyperwallet/clientsdk/model/HyperwalletUser.java:18]: Why are there changes to the existing enums like Gendre, ProfileType?
Why are they no longer static?

File: src/test/resources/integration/updateUsertaxVerificationStatus-response.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "token": "usr-b8e7ff1d-a3c6-45a0-ae0a-62b74580caca",
```
Comment (hrandhawa-hw) [inline src/test/resources/integration/updateUsertaxVerificationStatus-response.json:2]: A `t` in the file name is lower case when it should be upper case.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java [hunk 3]
```diff
@@ -223,6 +224,47 @@ public void testUpdateUserVerificationStatus() throws Exception {
...
+        try {
+            returnValue = client.updateUser(hyperwalletUser);
+        } catch (Exception e) {
+            mockServer.verify(parseRequest(functionality));
+            throw e;
+        }
+
+        assertThat(returnValue.getToken(), is(equalTo("usr-b8e7ff1d-a3c6-45a0-ae0a-62b74580caca")));
+        assertThat(returnValue.getStatus(), is(equalTo(HyperwalletUser.Status.PRE_ACTIVATED)));
+        assertThat(returnValue.getTaxVerificationStatus(), is(equalTo(TaxVerificationStatus.VERIFIED)));
```
Comment (hrandhawa-hw) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:246]: If the taxVerificationStatus is being updated to REQUIRED why is the response returning VERIFIED? Shouldn't it be REQUIRED?

Comment (hlahlou-pp-dev) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:246]: I believe this test is testing the ability to update a user from Required -> Verified, the above step is the initialization of arguments before running the test

**Comment by hlahlou-pp-dev** [comment]
Everything LGTM, just address the comments Harnek mad

**Review by hlahlou-pp-dev — APPROVED** [review]

File: src/test/resources/integration/updateUsertaxVerificationStatus-response.json (hunk 2 above)
Comment (saikgupta) [reply src/test/resources/integration/updateUsertaxVerificationStatus-response.json]: done

**Review by hrandhawa-hw — APPROVED** [review]

**Pull request merged**